airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -168,6 +168,9 @@ class KubernetesPodOperator(BaseOperator):
     :param labels: labels to apply to the Pod. (templated)
     :param startup_timeout_seconds: timeout in seconds to startup the pod.
     :param get_logs: get the stdout of the container as logs of the tasks.
+    :param log_containers: list of container names or bool value to collect logs.
+        If bool value is True, all container logs are collected,
+        if False, only 'base' container logs are collected.
```

Comment:
The False case feels unnecessarily complicated to me; I’d expect `log_containers=False` to not log anything. I’d make this take

1. `list[str]`: A list of container names
2. `str`: One single container name
3. `Literal[True]`: Log everything

and make the argument default to `BASE_CONTAINER_NAME`.

Reply:
Hi @uranusjr 
Thanks for reviewing the code. 

I have implemented 'log_containers' based on this [discussion](https://github.com/apache/airflow/pull/27802#issuecomment-1321574467). I can make these changes, but can I assume that this will be the final design?

@uranusjr @dstandish @dimberman @potiuk @bdsoha @XD-DENG

Reply:
> but can I assume that this will be the final design?

No. you can't. There is always possibility that others will come with more comments or find something else. So you have no gurarantees. It happens that I have to (even as maintainer) rewrite and fix and split and merge the same PR 10-20 times. This is possible.

I am not saying this will happen in this case. Just that you have no guarantees that there will be no more comments. You have to deal with this. And treat it as  a learning opportunity. I - for one - I am super happy when someone suggest a change that allows to remove half of the code I wrote and iterated over in a PR of mine even if it's done after weeks of iterating. 

The end result is better, often the comment would not even come if I did not iterate multiple times in the process. I've learned somethings. Only pluses.

Reply:
I think @uranusjr 's suggestion makes sense.  

But maybe we can avoid adding another parameter since there's already a get_logs param.

Current behavior of get_logs is base cont if True or none if false.

We could change meaning of True to get all containers.  This would be change in behavoir but nonbreaking.  

Aternatively, we could leave True to mean base container and add a sentinel value like ALL_CONTAINERS to get all containers.  WDYT @mlnsharma @uranusjr

Reply:
> but can I assume that this will be the final design?

and 

> No. you can't. There is always possibility that others will come with more comments or find something else

Maybe the sentiment expressed here is really, "before I make this change, what do others think? do you agree with this path forward?" :) 

It's a reasonable thing to ask.  And I understand it's hard as contributor sometimes to wait for reviews and sometimes get conflicting requests and this kind of thing.

Reply:
Sometimes those committers have 100 of PRs to look like. I suggest you to follow the request you got, otherwise you might wait for a long time. In the meantime after you do it it, maybe another commiter will have a look and have another idea. But you can follow what you got now. Or wait (sometimes weeks) for other's comment. 

It's enough to have approval of single commiter to get things merged in non-critical code.

Reply:
Good point, good point... i'm honestly still figuring out how all this should work too

Reply:
Hi @potiuk Thanks for the detailed response, I agree with your comments and hope to conclude it sooner by incorporating everyone's comments.

HI @uranusjr @dstandish Thanks for your reviewing my code & providing comments. As @dstandish commented, I can make use of the same 'get_logs' param to take the input. What do you suggest about having below options for the param value? 

- True: log all containers except airflow-xcom-sidecar container
- False: no logging of containers, as it is today
- list[str]: log containers specified by user

Apart from logging all containers, it also gives user an option to filter logging of containers. I hope this also makes it easier to understand. Please let me know your thoughts, I can make changes based on your comments.